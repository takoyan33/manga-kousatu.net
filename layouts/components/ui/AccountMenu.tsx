import ListSubheader from '@mui/material/ListSubheader'
import List from '@mui/material/List'
import ListItemButton from '@mui/material/ListItemButton'
import ListItemIcon from '@mui/material/ListItemIcon'
import DraftsIcon from '@mui/icons-material/Drafts'
import SendIcon from '@mui/icons-material/Send'
import Link from 'next/link'

export const AccountMenu = (onClick) => (
  <List
    sx={{ width: '100%', maxWidth: 300, bgcolor: 'background.paper' }}
    component='nav'
    aria-labelledby='nested-list-subheader'
    subheader={
      <ListSubheader component='div' id='nested-list-subheader'>
        アカウントメニュー
      </ListSubheader>
    }
  >
    <ListItemButton>
      <ListItemIcon>
        <SendIcon />
      </ListItemIcon>
      <Link href='/profile/edit'> プロフィールを変更する</Link>
    </ListItemButton>
    {/* <button className="m-5">
          <Link href="/profile/emailedit">メールアドレスを変更する</Link>
        </button> */}
    <ListItemButton>
      <ListItemIcon>
        <SendIcon />
      </ListItemIcon>
      <Link href='/profile/edit/password'>パスワードを変更する</Link>
    </ListItemButton>
    <ListItemButton>
      <ListItemIcon>
        <DraftsIcon />
      </ListItemIcon>
      <button className='' onClick={onClick}>
        アカウントを退会する
      </button>
    </ListItemButton>
  </List>
)
