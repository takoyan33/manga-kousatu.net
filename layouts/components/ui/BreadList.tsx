import Link from 'next/link'
import React from 'react'

interface BreadListParams {
  secondTitle: string
  secondUrl?: string
  thirdTitle?: string
  thirdUrl?: string
}

// eslint-disable-next-line react/display-name
export const BreadList = React.memo(
  ({ secondTitle, secondUrl, thirdTitle, thirdUrl }: BreadListParams) => {
    return (
      <div>
        <Link href='/'>トップ</Link>
        {secondTitle && (
          <span>
            ＞ <Link href={secondUrl || '/'}>{secondTitle}</Link>
          </span>
        )}
        {thirdTitle && (
          <span>
            ＞<Link href={thirdUrl || '/'}>{thirdTitle}</Link>
          </span>
        )}
      </div>
    )
  },
)
