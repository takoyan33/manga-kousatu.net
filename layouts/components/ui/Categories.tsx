export const Categories = [
  {
    id: 1,
    className: "bg-blue-500 hover:bg-blue-700",
    title: "ONEPIECE",
    link: "/post/categories/ONEPIECE",
  },
  {
    id: 2,
    className: "bg-purple-500 hover:bg-purple-700",
    title: "呪術廻戦",
    link: "/post/categories/呪術廻戦",
  },
  {
    id: 3,
    className: "bg-rose-500 hover:bg-rose-700",
    title: "東京リベンジャーズ",
    link: "/post/categories/東京リベンジャーズ",
  },
  {
    id: 4,
    className: "bg-yellow-500 hover:bg-yellow-700",
    title: "キングダム",
    link: "/post/categories/キングダム",
  },
];
