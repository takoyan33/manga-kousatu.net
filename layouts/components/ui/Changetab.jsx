import { useEffect, useState, useMemo } from "react";
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select, { SelectChangeEvent } from "@mui/material/Select";
import { database } from "../../../firebaseConfig";
import { collection, onSnapshot } from "firebase/firestore";
import { useRouter } from "next/router";
import { getAuth } from "firebase/auth";
import { query, orderBy } from "firebase/firestore";
import React from "react";

// type Props = {
//   getallPost?: () => Promise<void>;
//   getallOldpost?: () => Promise<void>;
//   getallLikepost?: () => Promise<void>;
// };

// eslint-disable-next-line react/display-name
export const Changetab = React.memo(
  ({ getallOldpost, getallPost, getallLikepost }) => {
    const [firedata, setFiredata] = useState([]);

    const databaseRef = collection(database, "posts");
    const q = query(databaseRef, orderBy("timestamp", "desc"));
    //新しい順
    const u = query(databaseRef, orderBy("timestamp"));
    //古い順
    const f = query(databaseRef, orderBy("likes", "desc"));
    //いいね数順

    return (
      <>
        <FormControl sx={{ m: 1, minWidth: 120 }} size="small">
          <InputLabel id="demo-select-small">並び順</InputLabel>
          <Select
            labelId="demo-select-small"
            id="demo-select-small"
            label="並び順"
          >
            <MenuItem value="新しい順" onClick={getallPost}>
              新しい順
            </MenuItem>
            <MenuItem value="古い順" onClick={getallOldpost}>
              古い順
            </MenuItem>
            <MenuItem value="いいね順" onClick={getallLikepost}>
              いいね順
            </MenuItem>
          </Select>
        </FormControl>
      </>
    );

    return { firedata };
  }
);
