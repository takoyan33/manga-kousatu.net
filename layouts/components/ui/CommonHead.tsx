import Head from 'next/head'
import React from 'react'

// eslint-disable-next-line react/display-name
export const CommonHead: React.VFC = React.memo(() => {
  return (
    <Head>
      <title>Manga Study</title>
      <meta name='description' content='Manga Study' />
      <link rel='icon' href='/favicon.ico' />
    </Head>
  )
})
