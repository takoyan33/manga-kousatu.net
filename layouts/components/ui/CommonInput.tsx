import TextField from '@mui/material/TextField'
import React from 'react'

type Props = {
  id: string
  text: string
  label: string
  error: boolean
  type: string
  helperText: string
  className: string
  variant?: 'standard' | 'outlined' | 'filled' | undefined
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
}

// eslint-disable-next-line react/display-name
export const CommonInput = React.memo(
  ({ onChange, error, text, type, helperText, label, id, className, variant }: Props) => {
    return (
      <>
        <div>
          <label className='my-4 text-center'>{text}</label>
        </div>
        <TextField
          id={id}
          type={type}
          label={label}
          className={className}
          variant={variant}
          onChange={onChange}
          error={error}
          helperText={helperText}
        />
      </>
    )
  },
)
