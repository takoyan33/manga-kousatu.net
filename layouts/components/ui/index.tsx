export * from "./Categories";
export * from "./Formcontrols";
export * from "./Profileid";
export * from "./Profilepost";
export * from "./SiteHead";
export * from "./SiteInput";
