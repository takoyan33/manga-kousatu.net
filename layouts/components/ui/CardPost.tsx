import React from 'react'
import Link from 'next/link'
import { useEffect, useState, useMemo, useCallback } from 'react'
import { database } from 'firebaseConfig'
import { collection, getDocs } from 'firebase/firestore'
import { useRouter } from 'next/router'
import { useGetPost, useGetUsers, useGetOtherUser } from 'layouts/components/hooks'
import { getAuth } from 'firebase/auth'
import styles from 'styles/Home.module.css'
import { Card, CardContent, Typography, Avatar } from '@mui/material'
import Image from 'react-image-resizer'
import Category from '../text/Category'
import FavoriteIcon from '@mui/icons-material/Favorite'
import { SingleGetPostParams } from 'types/post'
import { GetUser } from 'types/user'

// eslint-disable-next-line react/display-name
export const CardPost = React.memo(
  ({
    downloadURL,
    id,
    likes,
    title,
    category,
    netabare,
    displayName,
    email,
    photoURL,
    createTime,
    selected,
    userid,
  }: SingleGetPostParams) => {
    const [users, setUsers] = useState<GetUser>()
    const [comments, setComments] = useState('')
    const cardStyle: React.CSSProperties = {
      margin: '10px',
    }

    // const useGetPostComment = async () => {
    //   const commentseRef = collection(database, 'comments')
    //   const c = query(commentseRef, where('postid', '==', id))
    //   try {
    //     const querySnapshot = await getDocs(c)
    //     const allcomments = querySnapshot.docs.map((doc) => ({
    //       ...doc.data(),
    //       id: doc.id,
    //     }))
    //     console.log('allcomments', allcomments)
    //     setComments(allcomments)
    //   } catch (error) {
    //     console.log('Error fetching user data', error)
    //   }
    // }

    useEffect(() => {
      useGetOtherUser(setUsers, userid)
      // useGetPostComment()
    }, [])

    return (
      <div className='w-full cursor-pointer md:w-1/3'>
        <Link href={`/post/${id}`}>
          <div key={id} className='m-auto w-4/5'>
            <div className='m-auto my-2 hover:opacity-80' style={cardStyle}>
              <dl>
                <div className='m-auto flex justify-center'>
                  <Image
                    className='text-center'
                    height={150}
                    width={200}
                    src={downloadURL}
                    alt='画像'
                  />
                </div>
                <CardContent>
                  <h3 className='text-left text-xl font-semibold'>{title}</h3>
                  <div className='flex'>
                    <Category category={category} />
                    {netabare === 'ネタバレ有' ? (
                      <div>
                        <dl className='mx-1 mt-1 inline-block rounded-xl border border-red-500 p-1 text-center text-sm text-red-500'>
                          {netabare}
                        </dl>
                      </div>
                    ) : (
                      <dl className='mx-1 mt-1 inline-block rounded-xl border border-gray-700 p-1 text-center text-sm text-gray-500'>
                        {netabare}
                      </dl>
                    )}
                  </div>
                  <div className='m-auto w-80' style={styles}></div>
                  {/* <div className='mt-3 max-w-xs'>
                    {selected &&
                      selected.map((tag, i) => (
                        // <Link href={`/post/tag/${tag}`} key={i}>
                        <span
                          className='rounded-xl border border-cyan-700 py-1 px-2 text-center text-cyan-700'
                          key={i}
                        >
                          #{tag}
                        </span>
                        // </Link>
                      ))}
                  </div> */}
                  <div>
                    <div className='m-auto flex py-2'>
                      <dl>
                        <Avatar
                          className='m-auto max-w-sm border text-center'
                          sx={{ width: 30, height: 30 }}
                          alt='投稿者プロフィール'
                          src={users?.profileImage}
                        />
                      </dl>
                      <dl className='ml-2'>
                        {users?.userName}
                        <span className='ml- text-pink-400'>
                          <FavoriteIcon />
                        </span>
                        <span className='ml-1'>{likes}</span>
                      </dl>
                    </div>
                  </div>
                </CardContent>
              </dl>
            </div>
          </div>
        </Link>
      </div>
    )
  },
)
