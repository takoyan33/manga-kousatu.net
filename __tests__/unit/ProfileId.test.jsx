import { render, screen, fireEvent } from '@testing-library/react'
import { ProfileId } from '../../layouts/components/ui/ProfileId'
import '@testing-library/jest-dom'

describe('A1_Component > ProfileId', () => {
  it('[A1_5_1]レンダリングされているか', () => {
    render(
      <ProfileId id={1} profileImage='image' userName='aaaa' bio='aaaa' favorite={['onepiece']} />,
    )
  })

  it('[A1_5_2]引数のテスト', () => {
    const profileImage = 'https://example.com/image.jpg'
    const userName = 'テストユーザー'
    const bio = '自己紹介'
    const favorite = ['#漫画1', '#漫画2']

    render(
      <ProfileId
        id={1}
        profileImage={profileImage}
        userName={userName}
        bio={bio}
        favorite={favorite}
      />,
    )

    // 画像がレンダリングされているか確認
    const imageElement = screen.getByAltText('プロフィール画像')
    expect(imageElement).toHaveAttribute('src', profileImage)

    // 名前がレンダリングされているか確認
    const nameElement = screen.getByLabelText('name-text')
    expect(nameElement).toHaveTextContent(`${userName}`)

    // プロフィールがレンダリングされているか確認
    const bioElement = screen.getByLabelText('profile-text')
    expect(bioElement).toBeInTheDocument()
    expect(bioElement).toHaveTextContent(bio)

    // 好きな漫画がレンダリングされているか確認
    const favoriteElement = screen.getByLabelText('liked-text')
    expect(favoriteElement).toBeInTheDocument()
  })
})
