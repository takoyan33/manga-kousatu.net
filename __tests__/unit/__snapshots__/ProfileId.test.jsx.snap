// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A1_Component > ProfileId [A1_5_3] Pageのスナップショット 1`] = `
<div>
  <div
    class="m-auto flex max-w-md"
  >
    <div>
      <div
        class="MuiAvatar-root MuiAvatar-circular m-auto max-w-sm border text-center css-cxgc6-MuiAvatar-root"
      >
        <img
          alt="プロフィール画像"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="image"
        />
      </div>
    </div>
    <div>
      <p
        aria-label="name-text"
        class="m-5 text-lg"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AccountBoxIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
          />
        </svg>
         
        aaaa
      </p>
      <p
        class="m-5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FavoriteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
         好きな漫画
      </p>
      <div
        aria-label="liked-text"
        class="m-5"
      >
        <p
          class="my-2 text-cyan-700"
        >
          #
          onepiece
        </p>
      </div>
    </div>
  </div>
  <div
    class="m-auto my-10 max-w-md"
  >
    <p
      class="m-5"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mr-2 css-i4bv87-MuiSvgIcon-root"
        data-testid="BorderColorIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 24H2v-4h20v4zM13.06 5.19l3.75 3.75L7.75 18H4v-3.75l9.06-9.06zm4.82 2.68-3.75-3.75 1.83-1.83c.39-.39 1.02-.39 1.41 0l2.34 2.34c.39.39.39 1.02 0 1.41l-1.83 1.83z"
        />
      </svg>
       プロフィール
    </p>
    <p
      aria-label="profile-text"
      class="m-5"
    >
      aaaa
    </p>
  </div>
</div>
`;
